### PR TITLE
Use repl package and remove mobiledetect

### PR DESCRIPTION
**Pull request by markstory**
Instead of shipping the console command with the application skeleton using a separate package will allow us to deliver fixes to the console command in a reasonable way. It also enables us to provide more shortcut functions for use in the repl and development mode.

I've removed mobiledetect as a direct dependency. I don't think it gets a ton of usage by users and it can live on as a comment and optional package that can be installed as necessary.

### CODE REVIEW

File: src/Application.php [hunk 1]
```diff
@@ -128,8 +128,10 @@ public function services(ContainerInterface $container): void
     protected function bootstrapCli(): void
     {
         try {
+            $this->addPlugin('Cake/Repl');
```
Comment (othercorey) [inline src/Application.php:131]: Shouldn't this try one plugin at a time?

Comment (dereuromark) [reply src/Application.php:131]: In this case, yes.
The grouping only makes sense if the following can only exist or be available with the first.

File: src/Application.php [hunk 2]
```diff
@@ -128,8 +128,10 @@ public function services(ContainerInterface $container): void
...
             $this->addPlugin('Bake');
         } catch (MissingPluginException $e) {
+            debug($e->getMessage());
```
Comment (ADmad) [inline src/Application.php:134]: Including `config/bootstrap` file as done [here](https://github.com/cakephp/cakephp/blob/4ac765fdd47d6b185bf9c55dfd5bb068b3daba2c/phpstan.neon#L10) should make phpstan happy.

Comment (markstory) [reply src/Application.php:134]: Whooops this wasn't supposed to be here.

File: src/Application.php (hunk 1 above)
Comment (markstory) [reply src/Application.php:131]: I can split it up if you'd prefer.

Comment (othercorey) [reply src/Application.php:131]: A simple `foreach (['Cake/Repl', 'Bake'] as $plugin)` should work.

**Comment by ADmad** [comment]
>I don't think it gets a ton of usage by users...

A lot of those developing CMSs would need it, probably more than the repl feature :)

**Comment by markstory** [comment]
@ADmad Fair point I've put mobiledetect back.

File: config/bootstrap.php [hunk 3]
```diff
@@ -165,6 +165,11 @@
 
 /*
  * Setup detectors for mobile and tablet.
+ * To use these detectors run:
+ *
+ * `composer require "mobiledetect/mobiledetectlib:^2.8"`
+ *
+ * Before uncommenting this code.
```
Comment (garas) [inline config/bootstrap.php:172]: It is no more commented out.

File: src/Application.php [hunk 4]
```diff
@@ -127,10 +127,12 @@ public function services(ContainerInterface $container): void
...
-        try {
-            $this->addPlugin('Bake');
-        } catch (MissingPluginException $e) {
-            // Do not halt if the plugin is missing
+        foreach (['Cake/Repl', 'Bake'] as $plugin) {
+            try {
+                $this->addPlugin($plugin);
+            } catch (MissingPluginException $e) {
+                // Do not halt if the plugin is missing
+            }
```
Comment (raul338) [inline src/Application.php:135]:
```suggestion
        $this->addOptionalPlugin('Cake/Repl');
        $this->addOptionalPlugin('Bake');
```
IMHO Is cleaner

File: src/Application.php [hunk 5]
```diff
@@ -127,10 +127,8 @@ public function services(ContainerInterface $container): void
      */
     protected function bootstrapCli(): void
     {
-        try {
-            $this->addPlugin('Bake');
-        } catch (MissingPluginException $e) {
-            // Do not halt if the plugin is missing
+        $this->addOptionalPlugin('Cake/Repl');
+        $this->addOptionalPlugin('Bake');
         }
```
Comment (othercorey) [inline src/Application.php:132]:
```suggestion
        $this->addOptionalPlugin('Cake/Repl');
        $this->addOptionalPlugin('Bake');
```

**Pull request merged**